### DISCUSSION TITLE
upgrade from a very old version

### DISCUSSION THREAD

**Discussion by wdbartram** [angristan/openvpn-install#802] [Help]
Hi Guys

I have been using the script for a very long time THANKS.

Some of the newer featues look very interesting, could somebody help me with the steps requred to upgrade.

I dont want to loose my existing keys, i.e. ovpn files.

Thanks in advance

William

**Checklist**

- [x] I read the [README](https://github.com/angristan/openvpn-install/blob/master/README.md)
- [x] I read the [FAQ](https://github.com/angristan/openvpn-install/blob/master/FAQ.md)
- [x] I searched the [issues](https://github.com/angristan/openvpn-install/issues?q=is%3Aissue+)
- [x] My issue is about the script, and not OpenVPN itself

**Describe the issue**

**To Reproduce**

1. ...

**Expected behavior**

**Logs**
If applicable, add logs or screenshots to help explain your problem.

If you can reproduce the issue, please run the script in debug mode and post the output: `bash -x openvpn-install.sh`

**Server if applicable):**

- OS: [e.g. Debian 10]
- Hosting provider (if applicable): [e.g. Vultr, AWS]

**Client (if applicable):**

- Client: [e.g. OpenVPN Connect]

**Additional context**

**Comment by khouryap** [comment]
Just looking to see if you resolved this -- also looking to update my installation and curious what you found the best method to be?

**Comment by hellresistor** [comment]
save your key files , remove & purge openvpn with apt. then install new opevpn script.
If you have difficult to replace the key files on right place,  simulate a new client and check the skelleton of folders.
hope this help you. (I have done this some month ago, get NO problems ).

Or Just wait for an 'official' method

**Comment by ryan-seifi** [comment]
Hi, how can I manually purge client from that "Very old" version?
